src/domain/BaseEntity/BaseEntity.ts

```diff
@@ -0,0 +1,13 @@
+import crypto from 'crypto';
+
+export class BaseEntity {
+  id: string;
+  createdAt?: Date;
+  updatedAt?: Date;
+
+  constructor() {
+    this.id = crypto.randomUUID();
+    this.createdAt = new Date();
+    this.updatedAt = new Date();
```

Comment:
Isn't this problematic? If an instance extends BaseEntity the class that extends has to call super() and a new id will be generated even for an entity that might not need it.
This kind of functionality should be handled by the ORM

Reply:
Creating an id when calling new can be useful in testing. also, let's think of a situation where we call new() without needing the properties above.. from my point of view is not problematic. I am happy to change this approach if we do find a  problem tho